lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -830,11 +831,18 @@ def _validate_issuer():
                     )
 
         def _validate_has_expired():
-            if self.has_expired:
-                if self.has_expired != self.cert.has_expired():
-                    self.message.append(
-                        'Certificate expiration check failed (certificate expiration is %s, expected %s)' % (self.cert.has_expired(), self.has_expired)
-                    )
+            # The following 3 lines are the same as the PyOpenSSL code.
```

Comment:
```suggestion
            # The following 3 lines are the same as the current PyOpenSSL code for cert.has_expired().
```
Otherwise it's not clear which PyOpenSSL code you're talking about :)